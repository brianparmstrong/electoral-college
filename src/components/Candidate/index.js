import { number, string } from 'prop-types';
import ElectoralVotes from '../ElectoralVotes';
import * as assets from '../../assets';

const Candidate = ({
  name,
  party,
  popularVoteTotals,
  winnerTakeAllTotal
}) => {
  const src = assets[`${name.toLowerCase()}Img`];
  return (
    <div className='candidate' id={name}>
        <h2>{name}</h2>
        <img src={src} alt={name}/>
        <ElectoralVotes
          party={party}
          popularVoteTotals={popularVoteTotals}
          winnerTakeAllTotal={winnerTakeAllTotal} />
    </div>
  );
};

Candidate.propTypes = {
  name: string,
  party: string,
  popularVoteTotals: number,
  winnerTakeAllTotal: number
};

export default Candidate;
