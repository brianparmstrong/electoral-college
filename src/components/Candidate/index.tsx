import ElectoralVotes from '../ElectoralVotes';
import * as assets from '../../assets';
import { IfcCandidate } from '../../types';

const Candidate = (candidate: IfcCandidate) => {
  const { name, party, popularVoteTotals, winnerTakeAllTotal } = candidate;
  const src = assets[`${name.toLowerCase()}Img`];
  return (
    <div className="candidate" id={name}>
      <h2>{candidate.name}</h2>
      <img src={src} alt={name} />
      <ElectoralVotes
        party={party}
        popularVoteTotals={popularVoteTotals}
        winnerTakeAllTotal={winnerTakeAllTotal}
      />
    </div>
  );
};

export default Candidate;
