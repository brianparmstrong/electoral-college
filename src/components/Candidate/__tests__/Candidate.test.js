import React from 'react';
import renderer from 'react-test-renderer';
import Candidate from '../';

const MOCK_CANDIDATE = {
  name: 'Dem',
  party: 'dem',
  popularVoteTotals: 40,
  winnerTakeAllTotal: 48
};

it('renders correctly', () => {
  const candidate = renderer
    .create(
      <Candidate {...MOCK_CANDIDATE} />
    )
    .toJSON();
  expect(candidate).toMatchSnapshot();
});
