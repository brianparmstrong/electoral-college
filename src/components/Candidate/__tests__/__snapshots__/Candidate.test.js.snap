// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="candidate"
  id="Dem"
>
  <h2>
    Dem
  </h2>
  <img
    alt="Dem"
    src="demlogo.jpeg"
  />
  <div
    className="electoralVotesWrapper"
  >
    <p
      className="electoralVotesLabel"
    >
      Winner-Take-All:
    </p>
    <span
      className="electoralVotes"
      id="dem-wta"
    >
      48
    </span>
    <p
      className="electoralVotesLabel"
    >
      Proportional:
    </p>
    <span
      className="electoralVotes"
      id="dem-prop"
    >
      40
    </span>
  </div>
</div>
`;
